src/wallet/walletutil.cpp

```diff
@@ -60,10 +61,16 @@ std::vector<fs::path> ListWalletDir()
     const size_t offset = wallet_dir.string().size() + 1;
     std::vector<fs::path> paths;
     boost::system::error_code ec;
+    boost::system::error_code eci;
 
-    for (auto it = fs::recursive_directory_iterator(wallet_dir, ec); it != fs::recursive_directory_iterator(); it.increment(ec)) {
+    for (auto it = fs::recursive_directory_iterator(wallet_dir, ec); it != fs::recursive_directory_iterator(); it.increment(eci)) {
         if (ec) {
-            LogPrintf("%s: %s %s\n", __func__, ec.message(), it->path().string());
+            LogPrintf("%s: iterator: %s %s\n", __func__, ec.message(), it->path().string());
+            continue;
+        }
```

Comment:
6307dfa87e31aa30e884489d36c67688527bf8ca

If `recursive_directory_iterator` ctor sets an error code we should not iterate `it`. Mind replacing `continue` with `break`, or, even better, initializing `it` before the loop and early return in case of error?

Reply:
Yes, you are right. If it failed initializing `it` then there is nothing to iterate. Fixed.

Reply:
We can also use `ec` again as I moved initialization of `it` before the loop.